credential/base/src/main/java/org/wildfly/security/credential/Credential.java

```diff
@@ -89,8 +103,22 @@ default boolean verify(Evidence evidence) {
      * @return {@code true} if the evidence is verified, {@code false} otherwise
      */
     default boolean verify(Supplier<Provider[]> providerSupplier, Evidence evidence) {
+        return verify(providerSupplier, evidence, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Verify the given evidence.
+     *
+     * @param providerSupplier the provider supplier to use for verification purposes
+     * @param evidence the evidence to verify (must not be {@code null})
+     * @param hashCharset the character set to use when converting the password string to a byte array
```

Comment:
Since `hashCharset` isn't allowed to be `null` here, would be good to add `(must not be {@code null})` to the end here.